src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -34,22 +33,49 @@ namespace Microsoft.VisualStudio.InteractiveWindow
     /// <summary>
     /// Provides implementation of a Repl Window built on top of the VS editor using projection buffers.
     /// </summary>
-    internal class InteractiveWindow : IInteractiveWindow
+    internal class InteractiveWindow : IInteractiveWindow, IInteractiveWindowOperations
     {
         private bool _adornmentToMinimize;
 
-        // true iff code is being executed:
-        private bool _isRunning;
-        private bool _isInitializing;
+        private State _DO_NOT_USE_stateBackingField;
+        private State _state
```

Comment:
Perhaps wrap the State field in a private struct with a property that controls access, to avoid naming _DO_NOT_USE_.